src/GraphQL/ResolveFieldContext/ResolveFieldContextExtensions.cs

```diff
@@ -55,7 +56,12 @@ private static bool TryGetArgument(this IResolveFieldContext context, System.Typ
         }
 
         /// <summary>Determines if the specified field argument has been provided in the GraphQL query request</summary>
-        public static bool HasArgument(this IResolveFieldContext context, string argumentName) => context.Arguments?.ContainsKey(argumentName) ?? false;
+        public static bool HasArgument(this IResolveFieldContext context, string argumentName)
+        {
+            var isIntrospection = context.ParentType == null ? context.FieldDefinition.IsIntrospectionField() : context.ParentType.IsIntrospectionType();
+            var argumentNameConverted = isIntrospection ? argumentName : (context.Schema?.NameConverter.NameForArgument(argumentName, context.ParentType, context.FieldDefinition) ?? argumentName);
+            return context.Arguments?.ContainsKey(argumentNameConverted) ?? false;
+        }
```

Comment:
```suggestion
        public static bool HasArgument(this IResolveFieldContext context, string name)
        {
            var isIntrospection = context.ParentType == null ? context.FieldDefinition.IsIntrospectionField() : context.ParentType.IsIntrospectionType();
            var argumentName = isIntrospection ? name : (context.Schema?.NameConverter.NameForArgument(name, context.ParentType, context.FieldDefinition) ?? name);
            return context.Arguments?.ContainsKey(argumentName) ?? false;
        }
```

To be consistent with other APIs in this file..